R/main.R

```diff
@@ -434,11 +436,16 @@ main_do_workflow <- function(x) {
   }
 
   if (pull) {
-    git_pull(config$root)
+    if (is.null(ref)) {
```

Comment:
As this duplicates this code (https://github.com/vimc/orderly/blob/d1628a254ac67e044422fbbfb84c6a71afbe067d/R/main.R#L185-L192) can you factor into its own function please?